### DISCUSSION TITLE
Snapshot database Error - What is the good way to migrate ?

### DISCUSSION THREAD

**Discussion by elecrenPLB** [directus/directus#12567] [Archive]
Hello ! I am new to Directus.

I first installed Directus in local, to try it. I am using MySQL 5.7
I created all the content I needed, and everything worked fine. So I tried to deploy it on my server.
I first tried to dump the database and restore it on my server database. It didn't work, my collections was database only and it was folders instead of collections... It was impossible to use

So I checked out the documentation, and I noticed I could use snapshots : https://docs.directus.io/reference/cli/#migrate-schema-to-a-different-environment

I made a snapshot :

`npx directus schema snapshot ./snapshot.yaml`

Then I created a new Directus project from scracth on my server and I tried to apply the snapshot :

`npx directus schema apply --yes ./path/to/snapshot.yaml`

Everything was correctly listed (relationships...) but I got some errors : 

`18:39:45 🚨 Failed to create relation "Certifications.editeur"`
`18:39:45 🚨 Collection "Certifications" doesn't exist `
`Error: Collection "Certifications" doesn't exist`

Is there anything I am doing wrong ?
And what could causes the problems I got when restoring the dabatase using a dump file (with same .env file) ?

Thank you in advance

**Comment by rijkvanzanten** [comment]
It's very hard to tell what's going wrong here without a little more information on the exact setup and schema snapshot file. Would you be able to share the snapshot file that you generated so we can try it out?

**Reply by elecrenPLB** [reply]
I created 2 new projects from scratch with the exact same fields... but this time it worked perfectly so I don't really know how to reproduce it

I tried again and again yesterday and it never worked. I only changed the names of the databases this time to try from scratch and it worked

**Reply by rijkvanzanten** [reply]
Hmmmmm in that case it'll be especially hard to figure out what went wrong 👀
